### ISSUE TITLE
[Web Accessibility] - onDidGainAccessibilityFocus does not trigger on voiceover navigation focus, even with MergeSemantics.

### ISSUE THREAD

**Issue by magicwhale** [flutter/flutter#187255]
### Steps to reproduce

Reopening https://github.com/flutter/flutter/issues/186271

I tried the suggested solution of wrapping my onDidGainAccessibilityFocus button's Semantic node with MergeSemantics. I then used VoiceOver to navigate onto the button with the MergeSemantics and Semantic nodes, and the print statement I added still doesn't trigger.

https://dartpad.dev/?id=fd22c3599240b8144aa5b2a5b0ade017 

### Expected results

The print statement I added to onDidGainAccessibilityFocus should trigger and print to console.

### Actual results

Nothing prints.

### Code sample

https://dartpad.dev/?id=fd22c3599240b8144aa5b2a5b0ade017

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by saurabh-mirajkar** [comment]
@magicwhale Thanks for the report. This issue has been reopened since the original thread #186271 was closed. Labeling for `team-accessibility` for further investigation.

**Comment by hannah-hyj** [comment]
I tested the codesample with MergeSemantics and I saw the onDidGainAccessibilityFocus getting printed in console, can you check again?

**Comment by magicwhale** [comment]
This is what it looks like when I try navigating around the code sample using VoiceOver on Mac:

https://github.com/user-attachments/assets/adb0109b-e42c-4060-a35e-7b5016df52b2

**Comment by hannah-hyj** [comment]
Thank you for sharing the screen recording. I tested it as a MacOS desktop app and it worked.
 If the issue is  on dartPad, it's a web issue instead of MacOS issue.

**Comment by hannah-hyj** [comment]
`didGainAccessibilityFocus` is not supported on web. 

See comments in engine/src/flutter/lib/ui/semantics.dart

/// On the web, the DOM "focus" event is equivalent to
  /// [SemanticsAction.focus]. Accessibility focus is not observable from within
  /// the browser. Instead, the browser, based on the platform features and user
  /// preferences, makes the determination on whether input focus should be
  /// moved to an element and, if so, fires a DOM "focus" event. This event is
  /// forwarded to the framework as [SemanticsAction.focus]. For this reason, on
  /// the web, the engine never sends [didGainAccessibilityFocus].

I will close this issue as intended.
